pkg/reconciler/buildrun/resources/sources/http.go

```diff
@@ -18,40 +19,31 @@ func AppendHTTPStep(
 	cfg *config.Config,
 	taskSpec *tektonv1beta1.TaskSpec,
 	source buildv1alpha1.BuildSource,
-) {
-	// HTTP is done currently all in a single step, see if there is already one
-	httpStep := findExistingHTTPSourcesStep(taskSpec)
-	if httpStep != nil {
-		httpStep.Container.Args[3] = fmt.Sprintf("%s ; wget \"%s\"", httpStep.Container.Args[3], source.URL)
-	} else {
-		httpStep := tektonv1beta1.Step{
-			Container: corev1.Container{
-				Name:       "sources-http",
-				Image:      cfg.RemoteArtifactsContainerImage,
-				WorkingDir: fmt.Sprintf("$(params.%s-%s)", prefixParamsResultsVolumes, paramSourceRoot),
-				Command: []string{
-					"/bin/sh",
-				},
-				Args: []string{
-					"-e",
-					"-x",
-					"-c",
-					fmt.Sprintf("wget \"%s\"", source.URL),
-				},
-			},
-		}
-
-		// append the git step
-		taskSpec.Steps = append(taskSpec.Steps, httpStep)
+) error {
+	if source.HTTP == nil {
+		return fmt.Errorf("http information for source %s is not specified", source.Name)
 	}
-}
 
-func findExistingHTTPSourcesStep(taskSpec *tektonv1beta1.TaskSpec) *tektonv1beta1.Step {
-	for _, candidateStep := range taskSpec.Steps {
-		if candidateStep.Name == "sources-http" {
-			return &candidateStep
-		}
+	httpStep := tektonv1beta1.Step{
+		Container: corev1.Container{
+			Name:  fmt.Sprintf("source-%s", source.Name),
```

Comment:
We so far only had `source-default` and the hard-coded `sources-http`. If we now finally create the container name like this, do we need to do some sanitization on it like for the volume name (see https://github.com/shipwright-io/build/blob/328f8c035a23a908fee1623d91947a1b0767bf51/pkg/reconciler/buildrun/resources/sources/utils.go#L56) ?

Reply:
Yes - volume names need to meet the DNS label naming standard, which is pretty strict (63 chars, lowercase, etc.)